amethyst_core/src/timing.rs

```diff
@@ -156,12 +165,36 @@ impl Time {
         self.time_scale = multiplier;
     }
 
-    /// Indicates a fixed update just finished.
+    /// Restarts the internal fixed update accumulator to the desired fixed update delta time.
+    ///
+    /// This should only be called by the engine.  Bad things might happen if you call this in
+    /// your game.
+    pub fn start_fixed_update(&mut self) {
+        self.fixed_time_accumulator += self.delta_seconds;
+    }
+
+    /// Checks to see if we should perform another fixed update iteration, and if so, returns true
+    /// and reduces the accumulator.
+    ///
+    /// This should only be called by the engine.  Bad things might happen if you call this in
+    /// your game.
+    pub fn step_fixed_update(&mut self) -> bool {
+        match self.fixed_time_accumulator >= self.fixed_seconds {
+            true => {
+                self.fixed_time_accumulator -= self.fixed_seconds;
+                true
+            }
+            false => false,
+        }
+    }
+
+    /// Updates the interpolation alpha factor given the current delta time and accumulator.
     ///
     /// This should only be called by the engine.  Bad things might happen if you call this in
     /// your game.
     pub fn finish_fixed_update(&mut self) {
-        self.last_fixed_update += self.fixed_time
+        self.interpolation_alpha = self.fixed_time_accumulator / self.delta_seconds;
+        self.interpolation_alpha_time = secs_to_duration(self.interpolation_alpha);
```

Comment:
I believe alpha is unitless, you are dividing seconds by seconds. It's an interpolation factor in 0..1 range. There is no point in converting it to duration and naming it `seconds`.

Also this calculation should probably be `self.fixed_time_accumulator / sellf.fixed_seconds`.

Reply:
What is the purpose of interpolation_alpha_time? I don't understand what it would be used for. The alpha is the only thing needed as far as I know (the number between 0 and 1).

Reply:
I've updated it to as suggested.